src/pages/node-express/lesson2-a1.mdx

```diff
@@ -0,0 +1,306 @@
+
+```js
+const EventEmitter = require("events");
+
+const emitter = new EventEmitter();
+
+setInterval(() => {
+  emitter.emit("timer", "hi there");
+}, 2000);
+
+emitter.on("timer", (msg) => console.log(msg));
+```
+
+<p class="ml-8">Or, you could make an async function that waits on an event:</p>
+
+```js
+const EventEmitter = require("events");
+
+const emitter = new EventEmitter();
+
+const waitForEvent = () => {
+  return new Promise((resolve) => {
+    emitter.on("happens", (msg) => resolve(msg));
+  });
+};
+
+const doWait = async () => {
+  const msg = await waitForEvent();
+  console.log("We got an event! Here it is: ", msg);
+};
+
+doWait();
+emitter.emit("happens", "Hello World!");
+```
+
+<p class="ml-8">
+  Don't worry if that last one looks a bit complicated. That's expected as we
+  haven’t talked about creating promises yet.)
+</p>
+
+5. Change back to the `01-node-tutorial` directory and run `15-create-big-file.js`. This creates a big file in the content directory. You’ll note that the instructor has a `.gitignore` that includes the file name that's created by that code so that it isn’t stored in Github. You should also add the file (or files) that you wrote to in the previous steps to your `.gitignore` file.
+6. Now, change back to the `answers` directory, and write a program called `16-streams.js`. It should create a read stream for the big file (.`./content/big.txt`) with encoding of `"utf8"` and a `highWaterMark` of `200`. The `highWaterMark` is the maximum amount of bytes that node will read with each chunk of the stream. The program should initialize a counter to 0. Then it should handle the `“data”` event for the stream by incrementing the counter and logging the event result to the screen. Then it should handle the `“end”` event by reporting the number of chunks received. Finally, it should handle the stream `“error”` event by logging the error to the console. Test the program for several values of highWaterMark. You can look at `01-node-tutorial/16-streams.js` file to help you as needed.
+7. Have a look at `17-http-stream.js`. You don’t need to write a program, but observe how the chunks that are read from the stream are piped to the `res` object that is returned from the http request. Try to understand this program. Usually, if you are sending back voluminous data in response to an HTTP request, you want to break it up into chunks.
+
+That's all for this week, great job! Note, that while the `01-node-tutorial` folder has files for `13-event-emitter.js` and `14-request-event.js`, we've substituted those with the event emitter work in this week's assignment and the `prompter.js` work in the previous assignment. You can still feel free to follow along with the video and complete those files if you'd like.
```

Comment:
context on skipped files (maybe we also need to add a note in the reviewer notes for this week?)